pandas/core/arrays/datetimelike.py

```diff
@@ -823,25 +826,41 @@ def _validate_searchsorted_value(self, value):
         return value
 
     def _validate_setitem_value(self, value):
-        if lib.is_scalar(value) and not isna(value):
-            value = com.maybe_box_datetimelike(value)
 
         if is_list_like(value):
-            value = type(self)._from_sequence(value, dtype=self.dtype)
-            self._check_compatible_with(value, setitem=True)
-            value = value.asi8
-        elif isinstance(value, self._scalar_type):
-            self._check_compatible_with(value, setitem=True)
-            value = self._unbox_scalar(value)
+            value = array(value)
+            if is_string_dtype(value.dtype):
+                # We got a StringArray
+                try:
+                    # TODO: Could use from_sequence_of_strings if implemented
```

Comment:
So related to my comment above, this TODO item is not correct (also the "We got a StringArray" comment above is thus not correct)

Reply:
Do you have a suggestion for what to do here?  _from_sequence is clearly too permissive

Reply:
Why is `_from_sequence` too permissive for this purpose? It seems that whathever is allowed to create an array of this dtype, can also be allowed to set values?

Reply:
this gets back to your point of from_sequence being too permissive for DTA/TDA/PA; it would let through e.g. float64 or int64 which shouldnt allow

Reply:
This then becomes a bit ambiguous on what to allow and what not. Why allow ints to create but not to set, while allow strings in both creating and setting? 
So why not disallow setting with string as well?

I suppose the answer is: that is the current behaviour .. But so to come back to your question what to use here: if you only want to allow actual strings, and not eg integers in an object dtype array, you can actually use `infer_dtype` to check it are strings or datetime objects?

Reply:
Im assuming thats a rhetorical question; LMK if you actually are looking for an explanation.

So I'm clear: are you advocating for the status quo, or do you have something else in mind?

Reply:
Well, it's not really a rhetorical question, I actually find it inconsistent to allow certain data types in creation and others in setting (I am fine with being more strict in setting compared to creation, but then we should be strict altogether and also not allow strings in setting, IMO). 
But it's a question that doesn't need to be solved in this PR, that for sure!

To be clear: I am fine with the code, I only commented about the comment not really being correct. But now you updated the dtype check, I assume the comment is actually correct.

Reply:
> I actually find it inconsistent to allow certain data types in creation and others in setting

Yes.  The upshot is that the preferred way of minimizing inconsistencies is to make from_sequence more strict.  Let's leave that for another thread.